.github/workflows/ci.yml

```diff
@@ -31,6 +31,62 @@ jobs:
         with:
           name: SplunkOtel.xcframework
           path: ./SplunkRumWorkspace/SplunkRum/xcframeworks/SplunkOtel.xcframework
+  saucelabs-prepare:
+    runs-on: macOS-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+      - name: Build for Sauce Labs
+        run: |
+          BUILD_FOLDER="work/splunk-otel-ios/splunk-otel-ios/SauceLabsTests"
+          BUILD_PATH="Build/Products/Debug-iphonesimulator"
+          xcodebuild -showsdks
+
+          xcodebuild -workspace SplunkRumWorkspace/SplunkRumWorkspace.xcworkspace \
+            -scheme SauceLabsTestApp \
+            -configuration Debug \
+            -sdk iphonesimulator \
+            -derivedDataPath SauceLabsTests
+          zip ${GITHUB_WORKSPACE}/SauceLabsTestApp.zip $(find ~/$BUILD_FOLDER/$BUILD_PATH/SauceLabsTestApp.app -type f)
+      - name: Upload test app to Sauce Labs
+        id: sl-upload
+        run: |
+          cd /Users/runner/work/splunk-otel-ios/splunk-otel-ios/
+          item_id=$(curl -u "${{secrets.SAUCELABS_USER}}:${{secrets.SAUCELABS_KEY}}" --location \
+            --request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' \
+            --form 'payload=@"SauceLabsTestApp.zip"' \
+            --form 'name="SauceLabsTestApp.zip"' \
+            --form 'description="'"$GITHUB_SHA"'"' | jq -r .item.id)
+          echo "sl_file_id=$item_id" >> $GITHUB_OUTPUT
+    outputs:
+      sl_file_id: ${{ steps.sl-upload.outputs.sl_file_id }}
+  saucelabs-tests:
+    runs-on: macOS-latest
+    needs: saucelabs-prepare
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['14', '15', '16']
```

Comment:
This is great.  I really like that the test matrix is in github rather than in SauceLabs UI/config.